components/reference-implementation-openbanking-consumerdatastandards-au/src/main/java/org/wso2/openbanking/consumerdatastandards/au/extensions/handlers/CDSAuthorizationFlowHandler.java

```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    /**
+     * Handle persist authorized consent request to save the authorized consent and complete the consent flow.
+     * @param persistAuthorizedConsentRequestBody - the request body containing the consent data to be persisted
+     * @return Response containing the status of the persistence operation
+     */
+    public Response handlePersistAuthorizedConsent(PersistAuthorizedConsentRequestBody
+                                                           persistAuthorizedConsentRequestBody) {
+
+        try {
+            // Consent persist step
+            SuccessResponsePersistAuthorizedConsentData persistConsentData = CDSConsentAuthPersistUtil
+                    .CdsConsentPersist(persistAuthorizedConsentRequestBody);
+
+            SuccessResponsePersistAuthorizedConsent response = new SuccessResponsePersistAuthorizedConsent();
+            response.setResponseId(persistAuthorizedConsentRequestBody.getRequestId());
+            response.setStatus(SuccessResponsePersistAuthorizedConsent.StatusEnum.SUCCESS);
+            response.setData(persistConsentData);
+
+            return Response.status(Response.Status.OK).entity(new JSONObject(response).toString()).build();
+
+        } catch (Exception e) {
+            // Convert any other exceptions to CDSConsentException automatically, then handle
+            log.error("Unexpected error during consent persistence: " + e.getMessage(), e);
+
+            CDSConsentException cdsException = CDSConsentException.fromThrowable(e, 
+                "Consent persistence failed");
+            
+            AuthorizationFailureException authException = AuthorizationFailureException.createError(
+                cdsException.getErrorEnum(), cdsException.getErrorDetail(), 
+                CommonConstants.REJECTED_STATUS, persistAuthorizedConsentRequestBody.getRequestId()
+            );
+
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(authException.toFailedResponseJsonString()).build();
+        }
+    }
+
+    /**
+     * Handle pushed authorization request validation.
+     * @param validateAuthRequestBody
+     * @return Response200ForValidateAuthorizationRequest response
+     * @throws CDSConsentException
+     * @throws JsonProcessingException
+     */
+    public Response200ForValidateAuthorizationRequest handlePushedAuthorisationRequest(
+            ValidateAuthorizationRequestBody validateAuthRequestBody)
+            throws JsonProcessingException {
+
+        Response200ForValidateAuthorizationRequest response = new Response200ForValidateAuthorizationRequest();
+        String requestId = validateAuthRequestBody.getRequestId();
+        Object requestObject = validateAuthRequestBody.getData().getRequestObject();
+
+        // Validate PAR request object and capture the error response
+        JSONObject errorData = CDSPushedAuthRequestValidator.validateCdsPushedAuthRequest(requestObject);
```

Comment:
Shall we align to a common case for class, method naming and use it throughout the repository? Ether "CDS" or "Cds"